fast64_internal/oot/c_writer/oot_scene_bootup.py

```diff
@@ -0,0 +1,137 @@
+        + f"// #define BOOT_TO_SCENE\n"
+        + f"// #define BOOT_TO_SCENE_NEW_GAME_ONLY\n"
+        + f"// #define BOOT_TO_FILE_SELECT\n"
+        + f"#define BOOT_ENTRANCE 0\n"
+        + f"#define BOOT_AGE LINK_AGE_CHILD\n"
+        + f"#define BOOT_TIME NEXT_TIME_NONE\n"
+        + f"#define BOOT_CUTSCENE 0xFFEF\n"
+        + f"#define BOOT_PLAYER_NAME 0x15, 0x12, 0x17, 0x14, 0x3E, 0x3E, 0x3E, 0x3E\n\n"
+        + f"#endif\n"
+    )
+
+    writeFile(os.path.join(exportPath, "include/config/config_bootup.h"), data)
+
+
+def getParamsFromOptions(options: Any) -> tuple[str, str]:
+    timeOfDay = (
+        "NEXT_TIME_DAY"
+        if options.headerOption == "Child Day" or options.headerOption == "Adult Day"
+        else "NEXT_TIME_NIGHT"
+    )
+
+    linkAge = (
+        "LINK_AGE_ADULT"
+        if options.headerOption == "Adult Day" or options.headerOption == "Adult Night"
+        else "LINK_AGE_CHILD"
+    )
+
+    return timeOfDay, linkAge
+
+
+# converts ascii text to format for save file name.
+# see src/code/z_message_PAL.c:Message_Decode()
+def stringToSaveNameBytes(name: str) -> bytearray:
+    specialChar = {
+        " ": 0x3E,
+        ".": 0x40,
+        "-": 0x3F,
+    }
+
+    result = bytearray([0x3E] * 8)
+
+    if len(name) > 8:
+        raise PluginError("Save file name for scene bootup must be 8 characters or less.")
+    for i in range(len(name)):
+        value = ord(name[i])
+        if name[i] in specialChar:
+            result[i] = specialChar[name[i]]
+        elif value >= ord("0") and value <= ord("9"):  # numbers
+            result[i] = value - ord("0")
+        elif value >= ord("A") and value <= ord("Z"):  # uppercase
+            result[i] = value - ord("7")
+        elif value >= ord("a") and value <= ord("z"):  # lowercase
+            result[i] = value - ord("=")
+        else:
+            raise PluginError(
+                name + " has some invalid characters and cannot be used as a save file name for scene bootup."
+            )
+
+    return result
+
+
+class OOT_ClearBootupScene(bpy.types.Operator):
+    bl_idname = "object.oot_clear_bootup_scene"
+    bl_label = "Undo Boot To Scene"
+    bl_options = {"REGISTER", "UNDO", "PRESET"}
+
+    def execute(self, context):
+        clearBootupScene(bpy.path.abspath(context.scene.ootDecompPath))
+        self.report({"INFO"}, "Success!")
+        return {"FINISHED"}
+
+
+ootEnumBootMode = [("Play", "Play", "Play"), ("File Select", "File Select", "File Select")]
+
+
+class OOTBootupSceneOptions(bpy.types.PropertyGroup):
+    bootToScene: bpy.props.BoolProperty(default=False, name="Boot To Scene")
+    overrideHeader: bpy.props.BoolProperty(default=False, name="Override Header")
+    headerOption: bpy.props.EnumProperty(items=ootEnumHeaderMenuComplete, name="Header", default="Child Day")
+    spawnIndex: bpy.props.IntProperty(name="Spawn", min=0)
+    newGameOnly: bpy.props.BoolProperty(default=False, name="Override Scene On New Game Only")
```

Comment:
maybe a description could help clear this up
`description="Only use this starting scene after loading a new save file"`